awx/ui/client/lib/components/syntax-highlight/syntax-highlight.directive.js

```diff
@@ -0,0 +1,89 @@
+const templateUrl = require('~components/syntax-highlight/syntax-highlight.partial.html');
+
+function atSyntaxHighlightController ($scope, AngularCodeMirror) {
+    const vm = this;
+    const varName = `${$scope.name}_codemirror`;
+
+    function init () {
+        if ($scope.disabled === 'true') {
+            $scope.disabled = true;
+        } else if ($scope.disabled === 'false') {
+            $scope.disabled = false;
+        }
+        $scope.value = $scope.value || $scope.default;
+
+        initCodeMirror();
+
+        $scope.$watch(varName, () => {
+            $scope.value = $scope[varName];
+        });
+    }
+
+    function initCodeMirror () {
+        $scope.varName = varName;
+        $scope[varName] = $scope.value;
+        const codeMirror = AngularCodeMirror(!!$scope.disabled);
+        codeMirror.addModes({
+            jinja2: {
+                mode: $scope.mode,
+                matchBrackets: true,
+                autoCloseBrackets: true,
+                styleActiveLine: true,
+                lineNumbers: true,
+                gutters: ['CodeMirror-lint-markers'],
+                lint: true,
+                scrollbarStyle: null,
+            }
+        });
+        codeMirror.showTextArea({
+            scope: $scope,
+            model: varName,
+            element: `${$scope.name}_codemirror`,
+            lineNumbers: true,
+            mode: $scope.mode,
+        });
+    }
+
+    vm.name = $scope.name;
+    vm.rows = $scope.rows || 6;
+    if ($scope.init) {
```

Comment:
Is this a variable that CodeMirror consumes?  I can't find reference to it being used anywhere in these changes.